src/modules/test/main.c

```diff
@@ -0,0 +1,732 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <Common.h>
+#include <Module.h>
+
+#define DEFAULT_BIN_PATH "/usr/lib/osconfig"
+#define OSCONFIG_CONFIG_FILE "/etc/osconfig/osconfig.json"
+
+#define AZURE_OSCONFIG "Azure OSConfig"
+
+#define RECIPE_ACTION "Action"
+#define RECIPE_LOAD_MODULE "LoadModule"
+#define RECIPE_UNLOAD_MODULE "UnloadModule"
+#define RECIPE_MODULE "Module"
+
+#define RECIPE_TYPE "Type"
+#define RECIPE_REPORTED "Reported"
+#define RECIPE_DESIRED "Desired"
+#define RECIPE_COMPONENT "Component"
+#define RECIPE_OBJECT "Object"
+#define RECIPE_PAYLOAD "Payload"
+#define RECIPE_JSON "Json"
+#define RECIPE_STATUS "Status"
+#define RECIPE_DELAY "Delay"
+
+#define RECIPE_RUN_COMMAND "RunCommand"
+
+typedef enum STEP_TYPE
+{
+    MODULE = 0,
+    COMMAND,
+    TEST
+} STEP_TYPE;
+
+typedef enum PAYLOAD_TYPE
+{
+    DESIRED = 0,
+    REPORTED
+} PAYLOAD_TYPE;
+
+typedef struct TEST_STEP
+{
+    PAYLOAD_TYPE type;
+    char* component;
+    char* object;
+    JSON_Value* payload;
+    int status;
+} TEST_STEP;
+
+typedef enum MODULE_ACTION
+{
+    LOAD = 0,
+    UNLOAD
+} MODULE_ACTION;
+
+typedef struct MODULE_STEP
+{
+    MODULE_ACTION action;
+    char* name;
+} MODULE_STEP;
+
+typedef struct COMMAND_STEP
+{
+    int status;
+    char* arguments;
+} COMMAND_STEP;
+
+typedef struct STEP
+{
+    STEP_TYPE type;
+    int delay;
+    union
+    {
+        MODULE_STEP module;
+        COMMAND_STEP command;
+        TEST_STEP test;
+    } data;
+} STEP;
+
+static bool g_verbose = false;
+
+void FreeStep(STEP* step)
+{
+    if (step == NULL)
+    {
+        return;
+    }
+
+    switch (step->type)
+    {
+        case MODULE:
+            free(step->data.module.name);
+            break;
+        case COMMAND:
+            free(step->data.command.arguments);
+            break;
+        case TEST:
+            free(step->data.test.component);
+            free(step->data.test.object);
+            json_value_free(step->data.test.payload);
+            break;
```

Comment:
>break; [](http://example.com/codeflow?start=12&length=6)

This last break is not needed. Also, is best to have as the last label "default"